lib/finch/http2/pool.ex

```diff
@@ -33,37 +33,20 @@ defmodule Finch.HTTP2.Pool do
     timeout = opts[:receive_timeout]
     request_ref = make_request_ref(pool)
 
-    metadata = %{request: request}
-    start_time = Telemetry.start(:send, metadata)
-
-    with :ok <- :gen_statem.call(pool, {:request, request_ref, request, opts}) do
-      Telemetry.stop(:send, start_time, metadata)
+    with {:ok, recv_start} <- :gen_statem.call(pool, {:request, request_ref, request, opts}) do
```

Comment:
I don't love this, but I'm not sure there's a good way around it, since we need the `:recv` start time to emit an exception if the given `request_fun` raises in the response receive loop.